### PR TITLE
Connecting backend

### PR DESCRIPTION
**Pull request by StinaSoile**
Frontend hakee kuvat ja muut sarjakuvien tiedot backendista.
Backend toisessa repositoryssa ja pyörii osoitteessa https://matikkasarjis-backend.onrender.com.
package.jsonissa on vaihtoehtoiset scriptit lokaalin ja globaalin backendin käytölle kehitysvaiheessa.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -34,18 +28,11 @@ function App() {
           />
           <Route
```
Comment (sepi4) [inline src/App.tsx]: this can be in loop

Comment (sepi4) [reply src/App.tsx]: something like this
```
comicNames = ["siivetonlepakko", "velhontaloudenhoitaja"]
...
comicNames.map ...
...
```

File: src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// export const apiBaseUrl = "http://localhost:3000/api";
+// export const apiBaseUrl = "https://matikkasarjis-backend.onrender.com/api";
+// export const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+export const apiBaseUrl =
+  process.env.NODE_ENV === "production"
```
Comment (sepi4) [inline src/constants.ts]: maybe change to NODE_ENV to BACKEND_ENV

Comment (sepi4) [reply src/constants.ts]: if changed, need to change it in other places also, like in package.json

File: src/constants.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// export const apiBaseUrl = "http://localhost:3000/api";
```
Comment (sepi4) [inline src/constants.ts]: remove comments

File: src/components/ComicList.tsx [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+        setVelhonTaloudenhoitajaEtusivu(
+          `${apiBaseUrl}/images/velhontaloudenhoitaja/${lepakko}`
+        )
+      );
+  });
   return (
     <div className="list-container center">
-
       <div
         className="list-element-made-by-sergei"
```
Comment (sepi4) [inline src/components/ComicList.tsx]: need to make better name

File: src/components/ComicList.tsx [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
-import SiivetonLepakkoEtusivu from "../../src/assets/SiivetonLepakko/varsinainen-etusivu.png";
+import { useState, useEffect } from "react";
+import { apiBaseUrl } from "../constants";
+import comicService from "../services/comicService";
+
 import "./ComicPage.css";
 
 const ComicList = () => {
   const navigate = useNavigate();
+  const [SiivetonLepakkoEtusivu, setSiivetonLepakkoEtusivu] = useState("");
```
Comment (sepi4) [inline src/components/ComicList.tsx]: I whould have `comicFrontPages` variable. 

- get all front pages
- loop front pages

Separate useState for each comic is not very good to me.

**Review by sepi4 — CHANGES_REQUESTED** [review]

File: src/constants.ts (hunk 2 above)
Comment (StinaSoile) [reply src/constants.ts]: if changed, the code breaks. For some reason it doesn't get recognized.

File: src/components/ComicList.tsx (hunk 4 above)
Comment (StinaSoile) [reply src/components/ComicList.tsx]: Done.

File: src/components/ComicList.tsx (hunk 5 above)
Comment (StinaSoile) [reply src/components/ComicList.tsx]: I don't know if I changed it for better or for worse, please look at it again!

File: src/constants.ts (hunk 2 above)
Comment (sepi4) [reply src/constants.ts]: True. Looks like it is not available on client side browser. 
Looks like you need to use vite method of setting custom env variables. 

Here good link with examples. 
https://vitejs.dev/guide/env-and-mode

So you can set variables in terminal commands like this

image

image

Or (better (maybe)) to have separate .env files (different modes) something like in examples https://vitejs.dev/guide/env-and-mode#modes
 So `.env.dev` and `.env.prod` can be created. Each would have own variables.

File: src/components/ComicList.tsx (hunk 5 above)
Comment (sepi4) [reply src/components/ComicList.tsx]: How about something like this? 
image

Comment (sepi4) [reply src/components/ComicList.tsx]: `comicInfo` thing is not so important, made it just to be able to loop over many comics front page infos.
More important `promise.all`, if you have to make many ruequests it good to make them all at once not one by one.

**Review by sepi4 — CHANGES_REQUESTED** [review]
i left comments

**Review by sepi4 — APPROVED** [review]

**Pull request merged**